src/types/media-session.ts

```diff
@@ -89,6 +89,28 @@ export class GenesysCloudMediaSession extends MediaSession {
     }
   }
 
+  end (reason: JingleReasonCondition | JingleReason = 'success', silent = false) {
+    this.state = 'ended';
+    this.processingQueue.kill();
+
+    if (typeof reason === 'string') {
+      reason = { condition: reason };
+    }
+
+    if (!silent) {
+      this.send('session-terminate', { reason });
+    }
+
+    setTimeout(() => {
```

Comment:
Can you also add a comment in the code about why we are waiting before we close the peerconnection. I think it'll be helpful when we look at this someday and wonder what the heck we are doing here.